packages/universal-router-sdk/src/entities/protocols/pancakeswap.ts

```diff
@@ -184,23 +174,29 @@ async function addV3Swap(
 ): Promise<void> {
   for (const route of trade.routes) {
     const { inputAmount, outputAmount } = route
-    // console.log(amountOut, amountOut, 'bhbhbhbhbhbhhbhbhbvhbhb')
 
     // we need to generaate v3 path as a hash string. we can still use encodeMixedRoute
     // as a v3 swap is essentially a for of mixedRoute
-    const path = encodeMixedRouteToPath(
+    const path = SmartRouter.encodeMixedRouteToPath(
       { ...route, input: inputAmount.currency, output: outputAmount.currency },
       trade.tradeType === TradeType.EXACT_OUTPUT
     )
-    const amountIn = maximumAmountIn(trade, options.slippageTolerance).quotient.toString()
-    const amountOut = minimumAmountOut(trade, options.slippageTolerance).quotient.toString()
+    const amountIn = SmartRouter.maximumAmountIn(trade, options.slippageTolerance).quotient.toString()
+    const amountOut = SmartRouter.minimumAmountOut(trade, options.slippageTolerance).quotient.toString()
 
     const recipient = routerMustCustody ? ROUTER_AS_RECIPIENT : validateAndParseAddress(options.recipient!)
 
     // similar to encodeV3Swap only we dont need to add a case for signle hop. by using ecodeMixedRoutePath
     // we can get the parthStr for all cases
     if (trade.tradeType === TradeType.EXACT_INPUT) {
-      const exactInputSingleParams = [recipient, amountIn, amountOut, path, payerIsUser]
+      const exactInputSingleParams = [
+        recipient,
+        amountIn,
+        performAggregatedSlippageCheck ? 0n : amountOut,
+        ,
```

Comment:
?

Reply:
this is samme method as in smart router sdk. this is correct. im pretty sure this aggregated slippage check is to help make sure trade doesnt fail when there is more than one swap. amountOutMin is pretty much similar to slippage so '0n' here is just a fallback i reckon to allow slippage for full range if multiple trades tto prevent execution from failing.

Reply:
is only done on trades with 2 swaps,